crates/core/src/constraint_system/prove.rs

```diff
@@ -155,6 +155,33 @@ where
 	let mut writer = transcript.message();
 	writer.write_slice(table_sizes);
 
+	let mut oracles = oracles.instantiate(table_sizes)?;
+
+	// Prepare the constraint system for proving:
+	//
+	// - Trim all the zero sized oracles.
+	// - Canonicalize the ordering.
```

Comment:
FWIW, I think we should impose requirements on the well-formedness of the constraint system before proving, and that would include things like:

- the flushes are ordered by channel_id.
- constraint sets always have at least one oracle.
- etc